### PR TITLE
added missing install rule, removed some dead CMake code

### PR DESCRIPTION
**Pull request by gentryx**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1143,13 +1137,9 @@ configure_file("${hpx_SOURCE_DIR}/cmake/templates/hpxrun.py.in"
...
-  DIRECTORY ${CMAKE_BINARY_DIR}/bin/
-  DESTINATION hpx/bin/
-  COMPONENT core
-  FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
-                   GROUP_READ GROUP_EXECUTE
-                   WORLD_READ WORLD_EXECUTE
-  FILES_MATCHING PATTERN "hpxcxx" EXCLUDE)
+  FILES "${CMAKE_BINARY_DIR}/bin/hpxrun.py"
+  DESTINATION bin
+  COMPONENT core)
```
Comment (hkaiser) [inline CMakeLists.txt]: Why is this change necessary?

Comment (gentryx) [inline CMakeLists.txt]: The previous code did not install anything, so I deleted it. The new code installs hpxrun.py which I need for the LGD test.

Comment (sithhell) [inline CMakeLists.txt]: Doesn't this remove hpxcxx? This sounds wrong!

Comment (gentryx) [inline CMakeLists.txt]: Heller: look again at the original code. The rule I had removed actually excludes hpxcxx. There are two hpxcxx in the build tree. The original code specifically excluded this hpxcxx as it is configured for the build tree.

Comment (gentryx) [inline CMakeLists.txt]: If the diff isn't clear: please take a look at the individual commits. Together with the accompanying messages, they should be self-explaining.

Comment (hkaiser) [inline CMakeLists.txt]: > The previous code did not install anything, so I deleted it. The new code installs 
> hpxrun.py which I need for the LGD test.

Sure, what I meant is why did you remove the `FILE_PERMISSIONS` clause? Shouldn't scripts be made executable?

**Comment by gentryx** [comment]
Gotcha. I've re-added the permissions for hpxrun.py and also copied those over for hpxcxx (which was missing them, too).

**Comment by sithhell** [comment]
Perfect, Thanks a lot!
